src/xenia/kernel/xboxkrnl/xboxkrnl_io.cc

```diff
@@ -491,6 +488,9 @@ dword_result_t NtQueryInformationFile(
         // };
         assert_true(length == 56);
 
+        // Make sure we're working with up-to-date information
```

Comment:
Shouldn't this no longer be necessary with commit 4b27aa1fca3063b3c4b3ff47ed7a7870af98e131?

Reply:
I'm not really sure, mostly kept that there in case anything else besides SetFileInfo changes the file size, eg. Windows docs on NtWriteFile mention that it can extend it under some circumstances, but I don't know if that's the case on 360 (if so, why does Forza do all this weird stuff with QueryFileInfo/SetFileInfo?)

IMO it's probably best to keep it there just to be safe, but if you like I can remove it, pretty sure Forza should still work without it.

Reply:
Could you add this rational to a comment next to this then? For future reference.

Reply:
Sure thing, extended that comment with some more context + squashed it.

Reply:
I feel like nothing has changed with the last commit?

Reply:
I guess I could mention that stuff about NtWriteFile too, would something like this be better?

```
// Make sure we're working with up-to-date information, just in case the
// file size has changed via something other than NtSetInfoFile
// (eg. seems NtWriteFile might extend the file in some cases)
```

Reply:
Oh I just meant it seemed like the new force push didn't have any new actual changes compared to what it was previously.

That'd probably be a nice note, yes.

Reply:
Oh right lol, that was just a quick typo fix, NtSetFileInfo -> NtSetInfoFile.

Added the note about NtWriteFile & squashed again.